frontend/src/metabase-lib/parameters/utils/targets.ts

```diff
@@ -56,3 +60,17 @@ export function getTargetFieldFromCard(
   const field = getParameterTargetField(target, metadata, question);
   return field ?? null;
 }
+
+export function compareMappingOptionTargets(
```

Comment:
@kulyk this is the function that we talked about - let me know if you think the location/logic could be changed here.

Reply:
@oisincoveney please use `getParameterTargetField` to get `Field` instance

Reply:
Not sure but maybe you would need to extract the dimension part into `getParameterTargetDimension` or something like this. Please note that this function requires a `Question` instance and it's correct - otherwise some inner functions might not work properly